tools/clang/unittests/HLSL/DxilContainerTest.cpp

```diff
@@ -1444,6 +1445,146 @@ TEST_F(DxilContainerTest, CompileCSWaveSizeRange_CheckPSV0) {
 TEST_F(DxilContainerTest, CompileWhenOkThenCheckRDAT) {
   if (m_ver.SkipDxilVersion(1, 3))
     return;
+  const char *shader =
+      "float c_buf;"
+      "RWTexture1D<int4> tex : register(u5);"
+      "Texture1D<float4> tex2 : register(t0);"
+      "RWByteAddressBuffer b_buf;"
+      "struct Foo { float2 f2; int2 i2; };"
+      "AppendStructuredBuffer<Foo> append_buf;"
+      "ConsumeStructuredBuffer<Foo> consume_buf;"
+      "RasterizerOrderedByteAddressBuffer rov_buf;"
+      "globallycoherent RWByteAddressBuffer gc_buf;"
+      "float function_import(float x);"
+      "export float function0(min16float x) { "
+      "  return x + 1 + tex[0].x; }"
+      "export float function1(float x, min12int i) {"
+      "  return x + c_buf + b_buf.Load(x) + tex2[i].x; }"
+      "export float function2(float x) { return x + function_import(x); }"
+      "export void function3(int i) {"
+      "  Foo f = consume_buf.Consume();"
+      "  f.f2 += 0.5; append_buf.Append(f);"
+      "  rov_buf.Store(i, f.i2.x);"
+      "  gc_buf.Store(i, f.i2.y);"
+      "  b_buf.Store(i, f.i2.x + f.i2.y); }";
+  CComPtr<IDxcCompiler> pCompiler;
+  CComPtr<IDxcBlobEncoding> pSource;
+  CComPtr<IDxcBlob> pProgram;
+  CComPtr<IDxcBlobEncoding> pDisassembly;
+  CComPtr<IDxcOperationResult> pResult;
+
+  struct CheckResFlagInfo {
+    std::string name;
+    hlsl::DXIL::ResourceKind kind;
+    hlsl::RDAT::DxilResourceFlag flag;
+  };
+  const unsigned numResFlagCheck = 5;
+  CheckResFlagInfo resFlags[numResFlagCheck] = {
+      {"b_buf", hlsl::DXIL::ResourceKind::RawBuffer,
+       hlsl::RDAT::DxilResourceFlag::None},
+      {"append_buf", hlsl::DXIL::ResourceKind::StructuredBuffer,
+       hlsl::RDAT::DxilResourceFlag::UAVCounter},
+      {"consume_buf", hlsl::DXIL::ResourceKind::StructuredBuffer,
+       hlsl::RDAT::DxilResourceFlag::UAVCounter},
+      {"gc_buf", hlsl::DXIL::ResourceKind::RawBuffer,
+       hlsl::RDAT::DxilResourceFlag::UAVGloballyCoherent},
```

Comment:
Did you mean to add a resource using the ReorderCoherent flag?

Reply:
> Did you mean to add a resource using the ReorderCoherent flag?

The SM69 version below has the `ReorderCoherent` case. This is the restored `CompileWhenOkThenCheckRDAT` for pre-DXIL 1.9.